js/settings.js

```diff
@@ -2,22 +2,30 @@ var load = require('load');
 
 require.scopes.settings =(() => {
     var settings = {};
+    let isReady = false
 
     // external settings defines a function that needs to run when a setting is updated
     var externalSettings = {
         'httpsEverywhereEnabled': function(value){ isExtensionEnabled = value }
     };
 
     function init() {
-        buildSettingsFromDefaults();
-        buildSettingsFromLocalStorage();
-        registerListeners();
+        return new Promise ((resolve, reject) => {
+            buildSettingsFromDefaults();
+            buildSettingsFromLocalStorage(resolve);
+            registerListeners();
+        })
+    }
+
+    function ready () {
+        return isReady
```

Comment:
The `.ready()` function here should return a chainable promise, not the boolean `isReady` state so that consumers of this module can do `settings.ready().then(...)`

Please see https://github.com/duckduckgo/chrome-zeroclickinfo/blob/beta/js/indexed-db.js#L45-L54